### DISCUSSION TITLE
HTML Indentation?

### DISCUSSION THREAD

**Discussion by mitten** [getpelican/pelican#3426] [Q&A] [answered]
I have been working on translating my site over to Pelican (which I'm generally really enjoying), but I have a question about HTML indentation. When the output files are generated, all indentation is lost. Is there a setting I can set to have the generator leave the tabs in when it writes the HTML code? I'm using Jinja style codes, and I tried setting the JINJA_ENVIRONMENT trim settings to False, but that didn't help.

For example, this part of my template has indentation:
```
        <div role="banner">
            <a href="/" class="home-link"><img src="/theme/image/mitten-site.png" alt="Mitten"/></a>
        </div>
    
        <nav role="menu">
            <ul>
                <li><a href="/">Blog</a></li>
            </ul>
        </nav>

        <main>
            {% block content %}
            {% endblock %}
        </main>
etc…
```

But the generated HTML output comes out like this, all left justified:
```
<div role="banner">
<a class="home-link" href="/"><img alt="Mitten" src="/theme/image/mitten-site.png"/></a>
</div>
<nav role="menu">
<ul>
<li><a href="/">Blog</a></li>
</ul>
</nav>
<main>
<article class="postlist-item long">
<time><a class="postlist-link" href="posts/2024/05/to-laptop-or-not-to-laptop/">May 07, 2024</a></time>
<div>
etc…
```

I know it's just housekeeping, but I find it easier to check the source code when it's indented. Is it a Markdown setting? I looked through those but didn't really see anything that looked applicable. Have I missed something somewhere?

**Comment by mitten** [comment, marked as answer]
I kept investigating, and it appears to be caused by using the [Image-Process plugin](https://github.com/pelican-plugins/image-process). After I uninstalled that, the HTML comes out pretty again. Not sure how to fix this, but at least I know what the culprit was and where to look.

**Reply by frederik-elwert** [reply]
I also noticed that, and it appears that this is an effect of the HTML parser the plugin uses by default. You could try switching to html5lib for example, that solved the issue for me. See https://github.com/pelican-plugins/image-process#selecting-a-html-parser
